geode-core/src/test/java/com/gemstone/gemfire/cache/query/QueryWithBucketParameterIntgTest.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*
+ * QueryWithBucketParameterTest.java
+ * JUnit based test
+ *
+ * Created on April 13, 2005, 2:40 PM
+ */
+package com.gemstone.gemfire.cache.query;
+
+        import static org.junit.Assert.*;
+
+        import java.io.Serializable;
+        import java.util.*;
+
+        import org.junit.After;
+        import org.junit.Before;
+        import org.junit.Rule;
+        import org.junit.Test;
+        import org.junit.experimental.categories.Category;
+
+        import com.gemstone.gemfire.cache.*;
+        import com.gemstone.gemfire.cache.query.internal.DefaultQuery;
+        import com.gemstone.gemfire.internal.cache.LocalDataSet;
+        import com.gemstone.gemfire.internal.cache.PartitionedRegion;
+        import com.gemstone.gemfire.test.junit.categories.IntegrationTest;
+        import org.junit.rules.ExpectedException;
+
+/**
+ * @implNote This test class has been designed to test the query execution using the LocalDataSet.executeQuery
+ * where one of the parameters passed is the bucket set to be used.
+ * Different variation of hashset variables are passed to the function to check for errors.
+ * Created by nnag on 3/3/16.
+ */
+@Category(IntegrationTest.class)
+public class QueryWithBucketParameterIntgTest {
+
+    //Necessary data elements required for execution of the test
+//    Region region;
+//    QueryService qs;
+//    Cache cache;
+    DefaultQuery queryExecutor;
+    LocalDataSet lds;
+
+    /**
+     * @implNote This code is copied from BugJUnitTest.java. These elements are required to pass as parameters
```

Comment:
Would it be possible to do the implNote now and break out this into it's own file if it is used by multiple tests or maybe use an existing object that fits? Saves us from having to revisit and creating additional work in the future.